### PR TITLE
Explainable AI: An Introduction

### PR DESCRIPTION
**Pull request by collins-a**
Article on Explainable AI.
Files: index.md, hero image as hero.jpg

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tool:** [Grammarly](https://grammarly.com)
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [ ] Have you used Repl.it for your code snippets?

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by collins-a** [comment]
Fixes #410

File: articles/explainable-ai/index.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+I recently wrote an article on bias in machine learning and how to handle it. To get a better understanding on bias, check it out [here](/engineering-education/handling-bias-in-machine-learning/).
+
+#### Models are More Complex and Unpredictable
+AI methods are usually either simple or complex. Simple methods are often rule-based and explainable. For example, a [decision tree](https://towardsdatascience.com/decision-trees-in-machine-learning-641b9c4e8052) model. This is a tree-like graph that splits a data set based on differing conditions. It is popularly used in classification problems, though it is also used for regression tasks. This type of model is very easy to understand and visualize for the user.
+
+On the other hand, complex methods are designed without explainability in mind. These methods often provide greater performance than simple AI methods. They are not designed to be interpretable. The advancements of deep learning, coupled with an increase in computational power (accessible for the algorithms), makes these complex methods even more unpredictable. Explanations become foggier. A trade-off between explainability and performance emerges.
```
Comment (hectorkambow) [inline articles/explainable-ai/index.md:77]: I think this paragraph is key in understanding the topic of the article. 
I had to read it a couple times to get what it was trying to convey. 
Perhaps going into some details as to how and why they are not interpretable ( perhaps detailing the added layers of complexity) in using neural networks or something around those line - to make it clear to the reader as to why when it comes to more complex methods we lose the explainability.

Comment (collins-a) [reply articles/explainable-ai/index.md]: Hey @hectorkambow, noted. Made some changes, let me know if they are satisfactory

File: articles/explainable-ai/index.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+5. Challenges to achieving explainable AI.
+
+#### Prerequisites
+To get the most from this article a basic understanding of artificial intelligence concepts will be useful. To understand some basic machine learning concepts check out this [article](/engineering-education/supervised-learning-algorithms/). A few concepts of machine learning and deep learning are also covered [here](/engineering-education/automated-fake-news-detection/).
+
+#### A few useful terms
+**Black box model** – a data mining and machine learning model whose internals are not interpretable to humans. The model internals are unknown to the observer. It could also be a proprietary model.
+
+**Neural network** – a series of algorithms that are modeled on the human brain, used to identify underlying relationships in data.
```
Comment (lalith1403) [inline articles/explainable-ai/index.md:40]: Rewrite as: 'a series of algorithms modeled on the human brain used to identify underlying data relationships.' for conciseness.

File: articles/explainable-ai/index.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+author: collins-ayuya
+date: 2020-10-20T00:00:00-11:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/explainable-ai/hero.jpg
+    alt: Explainable AI example image
+---
+AI-powered systems have a lot of influence on our daily lives. A number of these systems are so sophisticated that little to no human intervention is required in their design and deployment. These systems make a lot of decisions for us every single day. They make a lot of crucial decisions that affect human lives (for example, in medicine, finance, law, defense). But how they do is often not explained.
```
Comment (lalith1403) [inline articles/explainable-ai/index.md:17]: 'They make a lot of crucial decisions that affect human lives' -> 'They make many decisions that affect human lives'
'But how they do is often not explained.' -> 'But how they do so is often not explained.'

File: articles/explainable-ai/index.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+To get the most from this article a basic understanding of artificial intelligence concepts will be useful. To understand some basic machine learning concepts check out this [article](/engineering-education/supervised-learning-algorithms/). A few concepts of machine learning and deep learning are also covered [here](/engineering-education/automated-fake-news-detection/).
+
+#### A few useful terms
+**Black box model** – a data mining and machine learning model whose internals are not interpretable to humans. The model internals are unknown to the observer. It could also be a proprietary model.
+
+**Neural network** – a series of algorithms that are modeled on the human brain, used to identify underlying relationships in data.
+
+**Recurrent Neural Network** - a neural network containing loops that allow information to be stored within the network.
+
+**Deep learning** – a subset of machine learning where neural networks learn from large quantities of data. Deep learning covers neural networks with more than two layers.
```
Comment (lalith1403) [inline articles/explainable-ai/index.md:44]: I don't think a three hidden layers model qualifies to be a deep learning model. We can not actually specify what qualifies as a deep learning model and what does not. Overall, if the model is dealing with a lot of data, it has to have a sufficient number of parameters to train well. It's a vague statement and I suggest we re-structure this.

File: articles/explainable-ai/index.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+Most traditional machine learning approaches can be categorized as black-box approaches. They would only predict who should receive or be denied financial services without any explanation. The implementation of explainable AI would explain why someone should receive or be denied financial services. This offers transparency to the process. It also makes the process more auditable. Legal compliance is also improved since decisions can be justified within the scope of the applicable laws.
+
+### Explainable AI Techniques/ Approaches
+To develop explainable systems, two main techniques are used; *ante-hoc and post-hoc*.
+
+#### Ante-Hoc Methods
+These techniques entail involving explainability into a AI model from the very beginning.
+
+##### Reverse Time Attention Model (RETAIN)
+To assist doctors with understanding AI software predictions, the [RETAIN](https://arxiv.org/abs/1608.05745) model was developed. It utilizes two recurrent neural networks, each having an attention mechanism. The two RNNs received patient hospital visit data. The attention mechanism was responsible for explaining the focus of the neural network and how a choice was influenced.
```
Comment (lalith1403) [inline articles/explainable-ai/index.md:109]: This section needs more information. How was the model used to assist the doctor? What was the problem the doctor had, and how was it solved by RETAIN?

File: articles/explainable-ai/index.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+[LIME](https://www.oreilly.com/content/introduction-to-local-interpretable-model-agnostic-explanations-lime/) is a popular post-hoc method that learns an interpretable model and attempts to explain its prediction. Only after a decision has been made will it provide a explanation. Here is how it works. LIME receives input, then generates a new dataset composed of refined data samples. The next step involves populating corresponding predictions that would have been made by a black-box model if the aforementioned samples would have been used as input.
+
+Next is the training of an interpretable model (such as regression models or decision trees). The model is trained on the new dataset to help explain changes in the key extracted features.
+
+Let’s go over an example to make it a bit clearer. Consider an image classification problem. The input is image data to be classified. Let’s take an image of a cat. LIME blacks out different parts of the input image of a cat. It feeds the resulting images through the interpretable model. This is to test the model on which perturbations on the images throw off the model's ability to predict that the image is of a cat.
```
Comment (lalith1403) [inline articles/explainable-ai/index.md:122]: A wonderful intuitive example. Loved it! 💯

File: articles/explainable-ai/index.md [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+
+LIME uses this to predict the reasoning behind the decision making of the model. Taking our cat image example, LIME would find that erasing parts of the cat's face makes it more difficult for the model to identify that it is a cat. As a result, this would show that classification of an image of a cat would highly depend on the face of the cat.
+
+##### Black Box Explanation through Transparent Approximation (BETA)
+BETA is a post hoc method that is linked to Interpretable Decision Sets. [Interpretable Decision Sets](https://cs.stanford.edu/people/jure/pubs/interpretable-kdd16.pdf) is a framework used to build highly accurate and interpretive predictive models. These are sets of “if-then” rules.
+
+BETA learns decision sets through which each rule explains part of the behavior of the model. The learning process yields a high degree of agreement between the model and the explanation. The process also yields low ambiguity and high interpretability. For a more technical explanation on BETA check out this [paper](https://arxiv.org/abs/1707.01154).
```
Comment (lalith1403) [inline articles/explainable-ai/index.md:129]: I did not understand the statement: ' The learning process yields a high degree of agreement between the model and the explanation' 

Can you include an example here? What are interpretable decision sets? What is the need for using them and what advantages do they offer?

**Review by lalith1403 — CHANGES_REQUESTED** [review]
I loved this article on Explainable AI. I definitely learned a lot. You have the perfect sense of igniting curiosity in the reader. A couple of explanations are needed to ensure maximum value to the reader. I also found a few grammatical errors on Grammarly. I suggest you rectify those once you update the article. 

Excited to see the updated article. Great job!!!

**Comment by collins-a** [comment]
Hey @lalith1403, I appreciate the positive comments!
I've made some changes/improvements. Let me know if more areas need a closer eye.

**Pull request merged**